src/Juvix/Compiler/Backend/Geb/Evaluator.hs

```diff
@@ -200,8 +202,8 @@ evalCase :: EvalEffects r => Case -> Sem r GebValue
 evalCase c = do
   vCaseOn <- eval $ c ^. caseOn
   case vCaseOn of
-    GebValueMorphismLeft leftArg -> apply (c ^. caseLeft) (leftArg ^. leftInjValue)
-    GebValueMorphismRight rightArg -> apply (c ^. caseRight) (rightArg ^. rightInjValue)
+    GebValueMorphismLeft leftArg -> evalExtendContext (leftArg ^. leftInjValue) (c ^. caseLeft)
+    GebValueMorphismRight rightArg -> evalExtendContext (rightArg ^. rightInjValue) (c ^. caseRight)
```

Comment:
I don't see why this changed. Is there a good reason in pro of keeping it? I personally found the usage of `apply` much clear than `evalExtendedContext`, so can you undo this, unless there is something wrong with it?

Reply:
We no longer expect a lambda. The semantics of `case` in GEB changed. Now it implicitly introduces a binder.

Reply:
Oh!

Reply:
Let me review this part again, I overlooked it.